File: BubbleSort.java
package Sort;
import java.lang.*;
import java.util.*;

class Bubble{
	public void display(int[] arr){
		for(int i = 0;i < arr.length;i++){
			System.out.print(arr[i]+"\t");
		}
		System.out.println();
	}
	public void bubbleSort(int[] arr){
		for(int i = 0;i < arr.length;i++){
			for(int j = 1;j < (arr.length-i);j++){
				if(arr[j-1] > arr[j]){
					int temp = arr[j-1];
					arr[j-1] = arr[j];
					arr[j] = temp;
				}
			}
		}
	}
}

class BubbleSort{
	public static void main(String[] args) {
		try{
			Scanner sc = new Scanner(System.in);
			System.out.println("How many elements in array");
			int num = sc.nextInt();

			if(num <= 0){
				System.out.println("Size of array should be greater than 0");
				System.exit(-1);
			}

			int[] arr = new int[num];

			for(int i = 0;i < num;i++){
				System.out.println("Enter "+(i+1)+" element");
				arr[i] = sc.nextInt();
			}

			Bubble bobj = new Bubble();
			bobj.display(arr);
			System.out.println("Sorted array");
			bobj.bubbleSort(arr);
			bobj.display(arr);
		}
		catch(Exception e){
			System.out.println(e);
		}
	}
}

File: BubbleSort.c
#include<stdio.h>
#include <stdlib.h>

void display(int *arr,int len){

	int i = 0;
	for(i = 0;i < len;i++){
		printf("%d\t",arr[i]);
	}
	printf("\n");
}

void bubble(int *arr,int len){

	int i = 0,j = 0,temp = 0;
	for(i = 0;i < len;i++)
	{
		for(j = 1;j < (len-i);j++)
		{
			if(arr[j-1] > arr[j])
			{
				temp = arr[j];
				arr[j] = arr[j-1];
				arr[j-1] = temp;
			}
		}
	}
}

int main()
{
	int n=0,i=0;
	int *arr = NULL;

	printf("\nHow many numbers\t:");
	scanf("%d",&n);

	if(n <= 0){
		printf("Number should be greater than 0\n");
		exit(-1);
	}

	arr = (int *)malloc(sizeof(int)*n);

	if(arr == NULL){
		printf("Unable to allocate memeory\n");
		exit(-1);
	}

	for(i=0;i<n;i++)
	{
		printf("\nEnter %d element",i+1);
		scanf("%d",&arr[i]);
	}
	
	display(arr,n);
	printf("\nAfter sorting\n");
	bubble(arr,n);
	display(arr,n);
	return 0;
}
